Update introduction.md

LightIsLife opened the pull request:
Provide entry-user level friendly introduction with a step-by-step introductions of Concepts, principles and first steps. PR is follow-up of forum discussions: https://community.openhab.org/t/introduction-in-docs-proposed-rewrite/62772  
fixes #845 
fixes #849 
only partially considers #838 
Signed-off-by: Markus Lipp <lipp_markus@gmail.com> (github: LightIsLife)

MyRaceData commented:
Markus
I think this is great and glad you caught the minimum Pi requirements addition. Comments in the forum have settle down to the point where I think we can consider the verbage/text stable. Everyone seems happy. That  was 'design by committee' at its finest, so many people with opinions perhaps not looking at the greater goal of making the docs easier for new folks and instead defending their opinion. It gives us both some insight into how making some simple changes to the docs to make things easier for newbies can turn into a blood bath of quarreling. Next we'll see what it takes to get this merged. Now that I've commented in this thread, I'm pretty sure I'll get notifications if/when you or someone posts here. There may be some comment or reviews before merge and I'll be ready to lend a hand. Thanks for taking the lead on this doc and pushing thru until all were happy. I feel you've made the first positive step at making the docs better for new users. Bravo!
Andy

Confectrician commented:
hey @MyRaceData just as information:

You can subscribe to a PR of your choice and get notfications without having to comment it.

![image](https://user-images.githubusercontent.com/7161177/50738160-54cbf300-11d1-11e9-83c5-4437cd827a83.png)

I think you have to open the desktop view for it.
Don't know if it works on mobile too.

@LightIsLife are we ready for review?
I wanted to give the community discussion some time before checking the PR. 🙂

MyRaceData commented:
Thanks Jerome

LightIsLife commented:
@MyRaceData thanks: but kudos go to @mstormi: https://community.openhab.org/t/new-docs-discussion-should-we-recommend-a-platform/62822/64 he was the one who caught my oversight of not specifying the raspi version.
@Confectrician ready when you are...the timing is a little tight: it was only in the forum for about 36h by now (actually about 24h before I posted the PR), but there was some good feedback; happy to wait a little longer though if you prefer.

MyRaceData commented:
Markus
I wouldn't consider the minimum Pi requirements an oversight on your part, this is a complex document and it covers a lot of material. As part of my initial plan to try to improve the documentation, this page was going to be one of my first targets. But my plan was to make changes section by section in the hope that I could make many small changes over time. (in hindsight perhaps not the best plan as I've learned continuity is important also) Your method was much more ambitious, kudos to you for pushing it thru to this point of making a pull request. 
One of my (many) concerns was the 'recommended' platform issue. That was why I started that thread. It got hairy but I also feel it drew a lot of valuable feedback. When a thread on the forum gets a lot of replies, it keeps the thread toward the top of the forum page longer, which in turn catches more people's eyes. It is a fast moving forum and threads can get buried fast.
I know little about the Pi  and had read many threads from folks trying to solve problems with the PI. A lot of those threads, once forum regulars had distilled down to the root cause, it was an older less powerful version of Pi. As such I'm glad the subject of 'minimum requirements' came up and resolved by people who know much more about the platform then myself. Thank you to @mstormi  and hoping he pops in here.
Markus, I also want to credit you for the idea of starting this thread:
[https://community.openhab.org/t/introduction-in-docs-proposed-rewrite/62772](url)
What I liked and had considered doing something similar myself was in the first post, you created a replica of the actual documentation page. Then as every forum user looked at the thread, they didn't have to bother going and finding that spot in the actual documentation. Excellent idea and I'm curious how you replicated the links ect. very good! Then as the discussion progresses, the first post can be edited to reflect changes and add links.
Point is, I think we should consider the a sort of 'boiler plate method' for working on the docs. Take each page, start a thread with a replica doc page in first thread. ect ect. ect. Kudos to you excellent work and enjoying collaborating on it with you.

LightIsLife commented:
Thanks again @MyRaceData you are too kind. 
And also here, I learned from others: in an earlier exercise (docs on type conversion and how to deal with states in rules) @rkoshak introduced me to this way of working. It is pretty effective as it reaches a larger audience than PRs and makes PRs cleaner.

As for the links: I just copied the current file out of github and rewrote it as necessary just keeping with markdown as formatting tool. This way, is easiest as I don't have to reformat links for the PR. The only reformatting that is needed to move from the forum to github is to make sure that for the PR each sentence starts with a new line. Unfortunately, the forum does not support this convention...but then it is not too much work. 

Looking forward to working with you on this and hopefully future PRs.

rkoshak commented:
I picked a bad few days to come down with the flu. Great improvements! This new version is WAY better.

I have a couple of nit picks. Feel free to address them or not. The Doc is really fine as is.

- Line 44 reads a bit awkwardly. I think I understand the intent but maybe something along the lines of "You may need to learn more about computers that you currently do." And then line 45 would start with "But this may not be a problem..."   I like the original but it took a couple of readings to get the meaning. Of course I'm hopped up on codine so I may not know what I'm reading.

- Line 70 I'd add "and not complete" to the end just to emphasize that the table is more than just a simplification, it's missing stuff too (like persistence).

- Line 142: The second already is a little repetitive. Drop the second one. 

- In relation to 142, is it worth adding a couple sentence telling the user how they can find out if they do already own a device compatible with OH? That is a sticking point brought up lots of times on the forum. It could be just a couple sentences along the lines of

"You can find out by browsing the list of addons (link) for any technologies or services you may recognize or if you already know you have a smart device, search the addons for the brand or technology used by that device."

Great work everyone! I'm sorry I missed the party! :-D

MyRaceData commented:
Rich, sorry you were ill, feel better!
Party it was... the 'recommended platform' issue spawned it's own thread on the forum and bloody hand to hand combat broke out, Thanks for comments

mstormi commented:
As a conclusion from the discussion on the forum I'd complement the HW recommendation with a warning like
"Be aware of a Pi's limitations with RAM and SD cards, though".
Eventually add or link to a longer explanation:
"If you also want to run memory-intensive applications beyond openHAB itself such as InfluxDB and Grafana, consider going for a platform with more RAM. And check [the OH forum](https://community.openhab.org/t/introduction-in-docs-proposed-rewrite/62772/12) to get on the safe side w.r.t. SD card wearout problems".

Yeah I know it's the introduction, always hard to find a tradeoff.

ThomDietrich commented on introduction.md:
```diff
@@ -6,41 +6,170 @@ layout: documentation
+
+For anyone undecided: many people find that the simplest way to experiment with openHAB is to get a [Raspberry Pi (Version 2  or 3)](https://raspberrypi.org) and install [openHABian]({{base}}/installation/openhabian.html); a "hassle-free openHAB setup". 
+While openHABian offers a streamlined and simplified way to get up and running quickly, it is a complete openHAB home automation system easily capable of automating your entire home.
+However, it is worth noting two main limitations for Raspberries. 
```
```suggestion
However, it is worth noting two potential limitations of Raspberry Pis. 
```

I read the sentence wrong at first. This might be clearer.

MyRaceData replied:
'main' limitation sounds to absolute, potential is much better

LightIsLife replied:
works for me

ThomDietrich commented on introduction.md:
```diff
@@ -6,41 +6,170 @@ layout: documentation
+You can install openHAB on your desktop computer for evaluation purposes if you already have any of these systems available for use, but we recommend using a dedicated system in the long run. 
+If you feel serious about home automation it may be better to start with a dedicated system right away.
+
+For anyone undecided: many people find that the simplest way to experiment with openHAB is to get a [Raspberry Pi (Version 2  or 3)](https://raspberrypi.org) and install [openHABian]({{base}}/installation/openhabian.html); a "hassle-free openHAB setup". 
```
>openHABian - the hassle-free... 

?

MyRaceData replied:
The term 'hassle free' has been drawing some fire on the user forum in threads discussing this topic. See thread
(https://community.openhab.org/t/new-docs-discussion-should-we-recommend-a-platform/62822/73)
In fact it seems you are credited as the author - see post 73

ThomDietrich commented on introduction.md:
```diff
@@ -6,41 +6,170 @@ layout: documentation
+If you feel serious about home automation it may be better to start with a dedicated system right away.
+
+For anyone undecided: many people find that the simplest way to experiment with openHAB is to get a [Raspberry Pi (Version 2  or 3)](https://raspberrypi.org) and install [openHABian]({{base}}/installation/openhabian.html); a "hassle-free openHAB setup". 
+While openHABian offers a streamlined and simplified way to get up and running quickly, it is a complete openHAB home automation system easily capable of automating your entire home.
```
I'm not sure what you want to say with "While... it is a complete openHAB home automation system". Instead maybe mention that it is based on a normal Linux distribution thus helping users from the beginning to the full-blown setup

LightIsLife replied:
```suggestion
openHABian offers a streamlined and simplified way to get up and running quickly and provides a complete openHAB home automation system easily capable of automating your entire home.
```

LightIsLife replied:
Fixing sentence, hope that is acceptable for both of you

ThomDietrich commented on introduction.md:
```diff
@@ -6,41 +6,170 @@ layout: documentation
+For anyone undecided: many people find that the simplest way to experiment with openHAB is to get a [Raspberry Pi (Version 2  or 3)](https://raspberrypi.org) and install [openHABian]({{base}}/installation/openhabian.html); a "hassle-free openHAB setup". 
+While openHABian offers a streamlined and simplified way to get up and running quickly, it is a complete openHAB home automation system easily capable of automating your entire home.
+However, it is worth noting two main limitations for Raspberries. 
+They are limited in the RAM (memory) and may not perform well when additional database and data visaulization programs are installed.
```
Remove "the" in front of RAM.

LightIsLife replied:
```suggestion
They are limited in RAM (memory) and may not perform well when additional database and data visaulization programs are installed.
```

LightIsLife replied:
Fixed, thanks

ThomDietrich commented on introduction.md:
```diff
@@ -6,41 +6,170 @@ layout: documentation
+While openHABian offers a streamlined and simplified way to get up and running quickly, it is a complete openHAB home automation system easily capable of automating your entire home.
+However, it is worth noting two main limitations for Raspberries. 
+They are limited in the RAM (memory) and may not perform well when additional database and data visaulization programs are installed.
+Running Raspberries from SD-cards only may result in system instabilities as these memory cards can degrade quickly under openHAB's use conditions. 
```
This is not true. Yes it is wise to think about backups and yes there have been users with bad luck. This is not the general case, otherwise someone needs to explain to me why the 11th iteration of the Raspberry Pi is still entrusted with an SD card 😉

mstormi replied:
> This is not true. Yes it is wise to think about backups and yes there have been users with bad luck. This is not the general case, otherwise someone needs to explain to me why the 11th iteration of the Raspberry Pi is still entrusted with an SD card 😉

Because Pis are build to learn programming. They are not meant to be 24x7 servers.
I know you don't believe in it and yes it is not a compelling result and it depends on a number of factors if it happens (too difficult to discuss in that here or in any introductory article), but for sure there's way too many people to have suffered from this than for this to be "not true" either. Also note that "use conditions" also include power outages which are frequent in HA and another way to crash SD cards.
I think this general warning covers both and should therefore remain in the docs as it stands now.
Omitting it would frustrate a number of people when their system crashes and they would attribute that to openHAB, not their Pi.
And yes, the install instruction should be reworked with a more extensive explanation, too.

LightIsLife replied:
I understand that the discussion can be more nuanced and complicated. This part is the very first introduction, we simply cannot overload it with too many technical details. This intro is also deliberately written to connect with real novices. Anyone with more expertise will just glance at it sigh and move on. I am agreeing with @mstormi and suggest to keep as is

mstormi commented:
it still reads "the RAM (memory)" and there's a typo in the next line (visaulization)
And two spaces instead of one in "2  or 3"
Ok nit-picking but hey better now than never

mstormi commented:
it still reads "the RAM (memory)" and there's a typo in the next line (visaulization)
And two spaces instead of one in "2  or 3"
Ok nit-picking but hey better now than never

LightIsLife commented:
@mstormi thought this committ https://github.com/openhab/openhab-docs/pull/850/commits/6573926e27f71c0fdb526e4e9cf5a38a9b16c367 would have fixed the "the" issue. Anyway, new commit fixes it hopefully, removed one double-space (could find only one) and corrected spelling of visualization....unless I missed still something. 

@MyRaceData I don't know how to actually do that (giving you write access). However, you can just add comments as necessary... or start a review and add proposed commits right there as ThomDietrich did. But then again, I am not sure if anyone can open reviews. 
One thought I would like to add however: in my daylife I develop routinely docs by committee and discussions about preferred style can go in circles very quickly. Seemingly everyone participating has a preferred style, yet in most cases these style preferences do overlap marginally only. So my preference would be to focus on need-to-have edits and let go of the nice-to-have ones. Having said that, I am not a native English speaker, if you are, you certainly have a leg up over me in this regard.

Confectrician commented:
I would do furhter improvements on other text parts in another PR in the future.
In the past i had the impression, that PRs here get a big overhead if you try to discuss too many parts parallel.

There's a point, where you ahve to say that's enough workload now.

So i would say, we are at this point here now.
I will look over the changes now too and we will leave out the discussion of other parts for upcoming PRs. 🙂 

BR
Jerome

MyRaceData commented:
I concur completely Jerome, this is a huge improvement to this introduction, lets get this one merged. Great job Markus!

Confectrician commented on introduction.md:
```diff
@@ -6,41 +6,170 @@ layout: documentation
+Home automation is fascinating and requires a considerable investment of your time. 
+Here are some key considerations especially for new users. To be successful, you will need to:
+
+* **Start slowly and one step at a time.**
```
I am not sure about the bold usage here.
It looks a bit misplaced and all of those points are highlighted anyway through the list.

LightIsLife replied:
Unbolded, no problem.

Confectrician commented on introduction.md:
```diff
@@ -6,41 +6,170 @@ layout: documentation
+For almost everything, there is more than one way in openHAB to achieve a goal or perform a task. 
+Initially, this can be frustrating and confusing, but it also gives great flexibility and does not mandate the use of certain devices or tools.
+
+You may need to learn more about computers that you currently do.
```
"You may need to learn more about computers **than** you currently do."

LightIsLife replied:
Thanks for catching this

Confectrician commented on introduction.md:
```diff
@@ -6,41 +6,170 @@ layout: documentation
+You may need to learn more about computers that you currently do.
+But this may not be a problem as you are reading the documentation of software that will enable you to do home automation yourself.
+
+Lastly, be prepared to start a new hobby: home automation. 
```
I would add a blank line between here and the next paragraph.

New hobby "home automation" -> *Time to breath* -> Here we go with additional information.

LightIsLife replied:
Great, done

Confectrician commented on introduction.md:
```diff
@@ -6,41 +6,170 @@ layout: documentation
+
+| Concepts | Meaning | More Information |
+| --- | --- | --- |
+| Bindings | are the openHAB component that provides the interface to interact electronically with devices | _see below_ |
```
Was the limitation to "Bindings" made an purpose?

I am asking because:
In fact Bindings are only a sub category off all add-ons.
Add-ons is the navigation title in paper ui. 
Add-ons is the used term for the website category.

I know that this aims to be a simplified overview, but i would think about using addons, to help people find their way in the website navigation/paper ui (and possibly other places as well).

LightIsLife replied:
Very self-centered approach: I remember my confusion about the various add-ons while I was still struggling to understand the basic concepts. Not sure how to include transforms, persistence, integrations, voice and actions without writing a whole para. I wanted to leave it just as a straight line: device connects via binding to thing, which connects to item, which I can see on sitemap. Adding the other add-ons means adding a whole bunch of side chains....happy to get pointers if I am overthinking it. Otherwise, it would be my preference to leave as is

Confectrician replied:
Fine for me, just wanted to talk about it.
We can change it anyway later, when someone has a better approach. :)

Confectrician commented on introduction.md:
```diff
@@ -6,41 +6,170 @@ layout: documentation
+You can install openHAB on your desktop computer for evaluation purposes if you already have any of these systems available for use, but we recommend using a dedicated system in the long run. 
+If you feel serious about home automation it may be better to start with a dedicated system right away.
+
+For anyone undecided: many people find that the simplest way to experiment with openHAB is to get a [Raspberry Pi (Version 2 or 3)](https://raspberrypi.org) and install [openHABian]({{base}}/installation/openhabian.html); a "hassle-free openHAB setup". 
```
Personally i would not recommend a PI 2 nowadays.
Was there a discussion about it? I didn't follow the complete thread.

MyRaceData replied:
Jerome:
I asked for a consensus in the user forum thread discussing this issue. @rkoshak has replied, await a few comments from the rest of the regulars but the proposed text seems to strike a good balance. I have zero experience with Pi so feel unable to comment on it's suitability as 'the' recommended platform and I was never sure if a recommendation should be made at all. I defer to those more experienced then me.

LightIsLife replied:
Agree with @Confectrician ....I understand that others may disagree, but then the intent here is to give a recommendations for those who do not yet know otherwise. Will change to "(ver 3 for best experience)". I have a few ver2 as mediaplayers....and goodness are they slow....so if a user has no other preferences, I have no idea why we would subject her/him to an experience with a ver2....power users will contest this....but then this section is not written for them, they know better anyway.....and will not be mislead by whatever this section states.

Confectrician commented on introduction.md:
```diff
@@ -6,41 +6,170 @@ layout: documentation
+Lastly, be prepared to start a new hobby: home automation. 
+Basic functions can be achieved in openHAB rather quickly, e.g., switch lights on at a certain time. 
+Others will require much more efforts and thoughts, e.g., how do I determine that someone is home, including guests, but not counting pets? 
+The openHAB forum is a great place to learn and discuss. 
```
L50 - L53 are a bit redundant compared to the community part..
Was there a background for mentioning this twice?

LightIsLife replied:
No real reason, but I remember feeling strong about this issue when I wrote it. deleted

LightIsLife commented:
I believe all issues from code review have been addressed, thanks! As explained in code review too, would prefer to leave the focus with regard to the add-ons on bindings in order to make it simple.

olemr commented:
Great job!

Just a few topy improvements suggestions:

The computer will only do what you tell **it** to do.

, but doing so will require substantial **effort** from you. (or: substantial effort on your part.)

high**-**level description (or: top-level description))

The main purpose of Bindings **is** to establish the connection between ..

LightIsLife commented:
Thanks @olemr, fixed it.
